arch/src/lib.rs

```diff
@@ -14,23 +14,24 @@ use std::result;
 
 #[derive(Debug, PartialEq)]
 pub enum Error {
+    #[cfg(target_arch = "x86_64")]
```

Comment:
Is this code only x86_64 specific? I am expecting the setup to be sort of the same for both x86 and x86_64 so I don't think we should be so restrictive with errors. I suspect that changes would be minor between the 2 which means that the 2 architecture (x86 & x86_64) can share one module and have maybe just few lines specific to each architecture.